var/spack/repos/builtin/packages/kaldi/package.py

```diff
@@ -76,16 +76,23 @@ def install(self, spec, prefix):
             mkdirp(prefix.bin)
             for root, dirs, files in os.walk('.'):
                 for name in files:
-                    if name.endswith(".so") or name.endswith(".cc") \
+                    if name.endswith("." + dso_suffix) \
+                            or name.endswith(".cc") \
                             or name.endswith(".pptx"):
                         continue
-                    if "configure" is name:
+                    if "configure" == name:
                         continue
                     if os.access(join(root, name), os.X_OK):
                         install(join(root, name), prefix.bin)
 
             mkdir(prefix.lib)
-            install_tree('lib', prefix.lib)
+            for root, dirs, files in os.walk('lib'):
+                for name in files:
+                    if name.endswith("." + dso_suffix):
+                        fname, _ = os.path.splitext(name)
```

Comment:
@jcftang it looks like name is guaranteed to end with `"." + dso_suffix` here, so the effect of 

```
fname, _ = os.path.splitext(name)
fpath = join(root, "{0}.{1}".format(fname, dso_suffix))
```

looks the same as

```
fpath = join(root, name)
```

or is there a case I'm overlooking?

Reply:
@scheibelp it's filtering out .dylib on macos and .so on linux, that's why there is the dso_suffix